amity_app/classes/amity.py

```diff
@@ -261,19 +229,26 @@ def add_person(self, name, role, wants_accommodation="N"):
                     Amity.offices[random_room].allocations.append(staff_object.person_id)
                     Amity.staff[staff_object.person_id].has_office = random_room
 
-                    self.return_message += "and allocated to office " + random_room + "\n"
+                    return_message += "\n and allocated to the office " + random_room + "\n"
+                else:
+                    return_message += "\n and was not allocated an office due to lack of space\n"
+
+        else:
+
+            if role == "STAFF":
+
+                return_message += "\n But was not allocated an office due to lack of space\n"
+            else:
+                return_message += "\n But was not allocated an office or living space due to lack of space\n"
 
         if role == "FELLOW":
-            return [fellow_object.person_id, self.return_message]
+            return [fellow_object.person_id, return_message]
         if role == "STAFF":
-            return [staff_object.person_id, self.return_message]
+            return [staff_object.person_id, return_message]
 
     # creates rooms in amity
     def create_room(self, room_type, room_names):
 
-        if room_type not in ["OFFICE", "LIVINGSPACE"]:
-            return "\nRoom type must be office or livingspace"
-
         successfully_created_rooms = []
```

Comment:
this variable is unnecessary. you are only using it to retrieve a len() as an identifier. you can use the len() of the dictionary.